samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -121,12 +128,33 @@ public void onProcessorChange(List<String> processors) {
         ScheduleAfterDebounceTime.DEBOUNCE_TIME_MS, () -> doOnProcessorChange(processors));
   }
 
-  public void doOnProcessorChange(List<String> processors) {
+  void doOnProcessorChange(List<String> processors) {
     // if list of processors is empty - it means we are called from 'onBecomeLeader'
-    generateNewJobModel(processors);
-    if (coordinatorListener != null) {
-      coordinatorListener.onJobModelExpired();
+    List<String> currentProcessorIds = getActualProcessorIds(processors);
```

Comment:
What is the expected behaviour when `currentProcessorIds` is empty? Do we still generate job model, create barrier .. etc